Add a Thumbnail Resizing scrollbar to the FileSystem Dock

AdeptusForge opened the pull request:
Added a HScrollBar to the FileSystemDock parallel to the file search bar
that controls the size of file thumbnails.

Allows users to adjust FileSystem thumbnail size on the fly, while 
continually updating the FileList view. Resizing lets users see more
files in smaller spaces, or fit more files on the same bar.

For folders/directories with large differences in file count, having 
to go into the editor settings was cumbersome, which has now been 
entirely streamlined to work within the editor.

*Bugsquad edit:* Closes https://github.com/godotengine/godot-proposals/issues/12469

AdeptusForge committed:
Thumbnail Resizing Bar

KoBeWi commented on editor/filesystem_dock.cpp:
```diff
@@ -4213,6 +4218,22 @@ FileSystemDock::FileSystemDock() {
 
+	thumbnail_scroll_container = memnew(VBoxContainer);
+	thumbnail_scroll_container->set_alignment(ALIGNMENT_CENTER);
+	thumbnail_scroll_container->set_tooltip_text("Change Thumbnail Size");
```
This should be set on the slider.
Also it would be nice if the tooltip included current value.

AdeptusForge replied:
Good idea honestly.

KoBeWi commented on editor/filesystem_dock.cpp:
```diff
@@ -4213,6 +4218,22 @@ FileSystemDock::FileSystemDock() {
+	file_list_thumbnail_scroll->set_min(16);
+	thumbnail_scroll_container->add_child(file_list_thumbnail_scroll);
+	file_list_thumbnail_scroll->hide();
+	file_list_thumbnail_scroll->connect("value_changed", callable_mp(this, &FileSystemDock::_update_thumbnail_size).unbind(1).bind(file_list_thumbnail_scroll->get_value()));
```
```suggestion
	file_list_thumbnail_scroll->connect("value_changed", callable_mp(this, &FileSystemDock::_update_thumbnail_size));
```
Pretty sure this does nothing. It wouldn't be working correctly if it did (you bound a constant value).

AdeptusForge replied:
Interesting. I think I might've misunderstood and thought it bound a reference rather than a constant value. Even weirder that it works despite it being there...

AdeptusForge committed:
Fixing all reviewed issues

Needed to keep the thumbnail text as an initial value, but it now updates in line with everything else when the user changes the value.

AdeptusForge committed:
Merge branch 'godotengine:master' into ThumbnailResizer

AdeptusForge committed:
Fixing Accidentally Obsolete'd Code

AdeptusForge commented on editor/filesystem_dock.cpp:
```diff
@@ -4014,6 +4015,12 @@ MenuButton *FileSystemDock::_create_file_menu_button() {
 
+void FileSystemDock::_update_thumbnail_size(int p_thumbnail_size) {
+	thumbnail_size_setting = p_thumbnail_size;
+	file_list_thumbnail_scroll->set_tooltip_text("Change Thumbnail Size : " + rtos(thumbnail_size_setting));
```
I have attempted to get this code to work 10 different ways, and none of them work. Running it in NOTIFICATION_TRANSLATION_CHANGED causes the function to run, but it is never translated, regardless of context. This is particularly strange because other instances of TTR in the same file work just fine. Even if my translated text is slapped directly next to other TTR translated text with no function call, it still refuses to translate. TTRC also doesn't work. When I went into the callstack, the translationServer said 'unable to parse all latin characters'.

I am convinced its something on my own machine, but without the ability to actually test it, I have no real way of running/testing a translation on it.

AdeptusForge committed:
Fixing VerticalCenter & Displaying 0 on first use

KoBeWi commented on editor/filesystem_dock.h:
```diff
@@ -167,6 +167,8 @@ class FileSystemDock : public VBoxContainer {
 	MenuButton *tree_button_sort = nullptr;
 
 	LineEdit *file_list_search_box = nullptr;
+	VBoxContainer *thumbnail_scroll_container = nullptr;
```
Unused.

KoBeWi commented on editor/filesystem_dock.cpp:
```diff
@@ -619,6 +619,8 @@ void FileSystemDock::_notification(int p_what) {
 
 			overwrite_dialog_scroll->add_theme_style_override(SceneStringName(panel), get_theme_stylebox(SceneStringName(panel), "Tree"));
+
+			_update_thumbnail_tooltip();
```
This should only be called on NOTIFICATION_TRANSLATION_CHANGED.

KoBeWi commented:
The slider value is not synced with settings. Changing slider need to update the editor setting and vice versa.

Repiteo commented:
Superseded by #119103; thank you for your contribution!
